### DISCUSSION TITLE
Updating 2 tables in a single request using rpc

### DISCUSSION THREAD

**Discussion by sbutler-gh** [supabase/supabase#2300] [Questions] [answered]
Following a user action, I'm trying to update two tables: a `projects` table and a `user` table.

This is what I put into the query box, and it successfully executes:

```
create function append_entry_to_projects_and_user (projectid int, entry jsonb, userid uuid, user_entry jsonb) 
returns void as 
$$
begin;
  update projects
  set log = coalesce(log, '[]'::jsonb) || to_jsonb(entry)
  where id = projectid;

  update users_prod
  set projects_joined = coalesce(projects_joined, '[]'::jsonb) || to_jsonb(projectid), projects_history = coalesce(projects_history, '[]'::jsonb) || to_jsonb(user_entry), log = coalesce(log, '[]'::jsonb) || to_jsonb(user_entry)
  where id = userid;

  end;
$$ 
language sql;
```

However, when I run that in the application and send data to the `append_entry_to_projects_and_user` rpc, I get the error:

`{hint: null, details: null, code: "0A000", message: "BEGIN is not allowed in a SQL function"}`

Can someone recommend a fix?

**Comment by sbutler-gh** [comment, marked as answer]
Strangely, I got it to work by simply removing the begin and end statements:

```
create function append_entry_to_projects_and_user (projectid int, entry jsonb, userid uuid, user_entry jsonb) 
returns void as 
$$
  update projects
  set log = coalesce(log, '[]'::jsonb) || to_jsonb(entry)
  where id = projectid;

  update users_prod
  set projects_joined = coalesce(projects_joined, '[]'::jsonb) || to_jsonb(projectid), projects_history = coalesce(projects_history, '[]'::jsonb) || to_jsonb(user_entry), log = coalesce(log, '[]'::jsonb) || to_jsonb(user_entry)
  where id = userid;
$$ 
language sql;
```
Is there anything functionally wrong with this?  Anything that I should be worried about using it?

**Reply by steve-chavez** [reply]
Hey @sbutler-gh,

> Is there anything functionally wrong with this? Anything that I should be worried about using it?

No, not at all. BEGIN/END are required when you use `language plpgsql`, for `language sql` they're invalid.

**Reply by sbutler-gh** [reply]
Thanks so much, @steve-chavez !
